components/clm/src/biogeochem/CNPhenologyBeTRMod.F90

```diff
@@ -1542,45 +1576,78 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
+                        t_season = .true.
+                     else                            ! no cold season
+                        p_season = .true.
+                     end if
+                  else                     ! precipitation seasonality
+                     p_season = .true.
                   end if
-                  if (ivt(p)==nscereal .or. ivt(p) == nscerealirrig) gddmaturity(p)=min(gdd020(p),hybgdd(ivt(p)))
+               else if (cvt(p) >= tcvt) then   ! temperature seasonality
+                  t_season = .true.
+               else                        ! no seasonality at all - tropics
+                  no_season = .true.
+               end if
 
-                  leafc_xfer(p) = 1._r8 ! initial seed at planting to appear
-                  leafn_xfer(p) = leafc_xfer(p) / leafcn(ivt(p)) ! with onset
-                  crop_seedc_to_leaf(p) = leafc_xfer(p)/dt
-                  crop_seedn_to_leaf(p) = leafn_xfer(p)/dt
+               if (kmo .eq. plantmonth(p)               .and. &
+                    gdd820(p) /= spval                  .and. &
+                    gdd820(p) >= gddmin(ivt(p))) then
+                  ! also require reasonable soil moisture to plant
+                  ! between 0.2 (dry) and 0.8 (wet) saturation in top soil
+                  ! layers
+                  if ( (wf2(c) .ge. minwet .and. wf2(c) .le. maxwet)  .and. &
+                     (t_season                                  .and. &
+                     t10(p) /= spval .and. a10tmin(p) /= spval  .and. &
+                     t10(p)     > planttemp(ivt(p))             .and. &
+                     a10tmin(p) > minplanttemp(ivt(p)))               &
+                                                                .or.  &
+                     (p_season                                  .and. &
+                     forc_rain(c) .gt. minrain/dt)                    & ! rain threshold to trigger
+                                                                .or.  & ! the beginnig of rain season
+                     (no_season)) then
+
+
+                     ! impose limit on growing season length needed
+                     ! for crop maturity - for cold weather constraints
+                     croplive(p)  = .true.
+                     cropplant(p) = .true.
+                     idop(p)      = jday
+                     harvday(p)   = NOT_Harvested
+                     harvdate(p)  = NOT_Harvested
+                     harvdate(p)  = NOT_Harvested
+
+                     ! go a specified amount of time before/after
+                     ! climatological date
+                     if (ivt(p)==nsoybean .or. ivt(p) == nsoybeanirrig) gddmaturity(p)=min(gdd1020(p),hybgdd(ivt(p)))
+                     if (ivt(p)==ncorn .or. ivt(p)==ncornirrig) then
+                        gddmaturity(p)=max(950._r8, min(gdd820(p)*0.85_r8, hybgdd(ivt(p))))
```

Comment:
same a previous comment: what do you think about defining new variables for `950._r8`,  `0.85_r8`, `150._r8`, and `1850._r8`?

Reply:
See above comment.